maven-content-package-plugin is not adding subPackages to the vault/filter.xml

cpilsworth opened the pull request:
Given a standard project creation from the `aem-project-archetype` using 

```
mvn archetype:generate  \
  -DarchetypeRepository=http://repo.adobe.com/nexus/content/groups/public/  \
  -DarchetypeGroupId=com.adobe.granite.archetypes  \
  -DarchetypeArtifactId=aem-project-archetype  \
  -DarchetypeVersion=10
```

After adding the `acs-aem-commons[content]`  subPackage (see [commit](https://github.com/cpilsworth/archetype-filter-bug/commit/d7e2c434aee4361505a03f1219105f8b5cbc66a7)),  I've found that whilst the package is embedding, it is not being installed.

The reason for this seems to be that the package is not being added to the `META-INF/vault/filter.xml` file.  Adding the subpackage zip manually to this file allows it to be deployed as expected.

Normally the `content-package-maven-plugin` would generate this, but it seems as though the maven-resources-plugin [execution](https://github.com/Adobe-Marketing-Cloud/aem-project-archetype/blob/master/src/main/archetype/ui.apps/pom.xml#L91) is replacing the filter.xml with the source original.  A respository containing the code exhibiting the issue can be found [here](https://github.com/cpilsworth/archetype-filter-bug).

cpilsworth commented:
This PR removes the `maven-resources-plugin` handling of the `META-INF/vault` folder that is overwriting the `filter.xml` that the `content-package-maven-plugin` is updating with details of subPackages.

rombert commented:
Sorry for the delay , I will try and find some time to review and test

cpilsworth commented:
No problem, if there is anything I can do to make the testing easier, let me know.

cpilsworth commented:
Just came around to testing this again on a new project, there is a change missing.  The change should also be in the ui.content module, as it is this that the content-package-maven-plugin generates filter.xml entries based on the subPackage definitions that is being overwritten.  Will test and commit the change to that module's pom.

pmminov commented:
hi @cpilsworth. I applied your file changes to a project of mine. Without those resource entries the filter.xml contents are overwritten entirely with the subpackage entry instead. So the only package deployed is the ACS commons package.

I opted for manually updating my filter with this

<filter root="/etc/packages/adobe/consulting/acs-aem-commons-content-2.5.0.zip" />

This is more of a thing for people willing to use ACS Commons with this archetype, but definitely looking forward to it being automatic

cpilsworth commented:
Hi @pmminov  - I've rebased this PR against the project and generated the a project from the archetype.  To the basic template project I've added the acs-aem-commons library.  I've checked the project including the build output into the following project:

Here is the [ui.apps filter.xml including the subPackage](https://github.com/cpilsworth/test-aem-archetype-fix/compare/oob-build-output?expand=1#diff-2af65fedd6dc805f6eeef59639b27b3bR8).  You can see that it includes the original [filter.xml](https://github.com/cpilsworth/test-aem-archetype-fix/blob/master/ui.apps/src/main/content/META-INF/vault/filter.xml) and the [filter entry for the acs-aem-commons package](https://github.com/cpilsworth/test-aem-archetype-fix/commit/6095a94cc008ee340547f8aebdfd002f2b45085e)

Does this fit with your findings?  If I've understood correctly, you're finding that the source controlled filter.xml is not being merged with the entries from the subPackages?

pmminov commented:
@cpilsworth yeah, those were my findings, but I found the error, forgot to add

 <filterSource>src/main/content/META-INF/vault/filter.xml</filterSource>

HitmanInWis commented:
Just ran across this issue right now.  Would love to see this change get pulled in, but looking at the diff it looks like this PR has a bunch of co-mingled updates from other issues.

cpilsworth commented:
Hmm, not sure what's happened here.  The changes can be found in [e3bce](https://github.com/Adobe-Marketing-Cloud/aem-project-archetype/pull/43/commits/e3bcedd598bac480a2dce11b054389d370a862fc).  I'll take a look at fixing up the branch for the PR tomorrow.

cpilsworth commented:
Created PR #67 to replace this as there were problems with the branch.

gopikrishna123 commented:
I am getting this error, can you please help me.... thanks in advance

[INFO] ------------------------------------------------------------------------
[INFO] Building Company - UI apps 0.0.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.6.1:clean (default-clean) @ company.ui.apps ---
[INFO] Deleting C:\symphony\company\ui.apps\target
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4:enforce (enforce-maven) @ company.ui.apps ---
[INFO] 
[INFO] --- maven-resources-plugin:2.7:resources (default-resources) @ company.ui.apps ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 60 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.2:compile (default-compile) @ company.ui.apps ---
[INFO] No sources to compile
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] company ........................................... SUCCESS [0.474s]
[INFO] Company - Core .................................... SUCCESS [4.138s]
[INFO] Company - UI apps ................................. FAILURE [0.712s]
[INFO] Company - UI content .............................. SKIPPED
[INFO] Company - Integration Tests Bundles ............... SKIPPED
[INFO] Company - Integration Tests Launcher .............. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 7.133s
[INFO] Finished at: Wed Jan 04 02:08:12 EST 2017
[INFO] Final Memory: 27M/355M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.day.jcr.vault:content-package-maven-plugin:0.0.24:check-signature (default-check-signature) on project company.ui.apps: The plugin com.day.jcr.vault:content-package-maven-plugin:0.0.24 requires Maven version 3.1.0 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginIncompatibleException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :company.ui.apps
